18F/identity-idp
# Build local copy of Digital Analytics Program

## 🛠 Summary of changes

Updates usage of [Digital Analytics Program](https://digital.gov/guides/dap/) to build and serve a local copy of the script.

**Why?**

- Serving from local copy reduces security risks inherent to loading scripts from third-party CDNs, even if the third-party is a trusted government source
   - Allows further strictening of content security policy
- Incorporating into existing build toolchain adds support for:
   - Subresource integrity ([see related Slack thread](https://gsa-tts.slack.com/archives/C04823NMA/p1723645054271099))
   - Optimized minification from bundler dead code elimination (see upstream https://github.com/digital-analytics-program/gov-wide-code/pull/118)
- Adds ability to fast-track patches to the code (see example upstream https://github.com/digital-analytics-program/gov-wide-code/pull/121 )
- Sets precedent for copying external assets for local use, hopefully 

## 📜 Testing Plan

Add to `config/application.yml`:

```
participate_in_dap: true
```

Verify analytics script loads successfully:

1. Go to http://localhost:3000
2. Verify on errors in local console, and that Google Analytics loads successfully. You should see some warnings about Third-party cookies being blocked in future versions of Chrome
